dom/devicestorage/nsDeviceStorage.cpp

```diff
@@ -3075,15 +3091,17 @@ nsDOMDeviceStorage::AddOrAppendNamed(Blob* aBlob, const nsAString& aPath,
 
   nsCOMPtr<nsIRunnable> r;
 
+// Ronin: absolute path are supported
+#ifdef MOZ_WIDGET_GONK
```

Comment:
This patch is likely to be temporary.

Absolute paths are transformed in a relative paths (relative to the storage path).
The problem is that currently there's a mismatch between storage paths (~/.mozilla/b2g/xession.error/storage) and download path for instance (~/Downloads).
Once the devicestorage is sorted out we can revisit this patch and see if it's still needed.